reporting_app/static/charts.js

```diff
@@ -1,328 +1,547 @@
+        data: aggregated_data.map(function(d){return [parseInt(d[field_name]), d['tat']]}),
+        yAxis: 0,
+        type: 'line',
+        color: '#FF4D4D',
+        marker: {
+            enable: true,
+            symbol: 'square',
+            fillColor: '#FF4D4D'
+        },
+        tooltip: {
+            pointFormatter: function(){return format_text_tat("", this.x, this.y, time_period, '', 'weeks');}
+        }
+    }
+    series = [series1, series2, series3, series4];
+
+    return Highcharts.chart('highchart_cont', {
+        chart: {
+            zoomType: 'x'
+        },
+        title: {
+            text: 'Turnaround time per ' + time_period,
+        },
+        xAxis: {
+            type: 'datetime',
+            title: {
+                text: 'Date'
+            }
+        },
+        yAxis: [
             {
-                'id': fields[i]['name'],
-                'label': fields[i]['title'] || field['name'],
-                'type': fields[i]['type'] || 'number'
+                title: {
+                    text: 'Turnaround time (weeks)'
+                }
+            },
+            { // Secondary yAxis
+                title: {
+                    text: 'Number of sample',
+                },
+                 opposite: true
             }
-        );
-    }
+        ],
+        series: series,
+        credits: false
+    });
+}
+
 
-    var k;
-    var i;
-    for (k in dict) {
-        var cells = [format_func(k)];
-        for (i=0; i<len; i++) {
-            cells.push({'v': dict[k][fields[i]['name']]});
+// Return the id of the checked radio button base on the button name
+function get_radio_value(radio_name){
+    var options = document.getElementsByName(radio_name);
+    for (var i = 0; i < options.length; i++) {
+        if (options[i].checked) {
+            return options[i].id;
         }
-        rows.push({'c': cells});
     }
-    return {'cols': cols, 'rows': rows}
 }
 
-// Format a date as a week
-function format_week(dateformat) {
-    var d = new Date(dateformat);
-    return {
-        'v': d,
-        'f': 'week ' + moment(d).format('w YYYY') + ' (' + moment(d).startOf('week').format('DD/MM') + ' -  ' + moment(d).endOf('week').format('DD/MM') + ')'
-    };
-}
 
-// Format a date as a month
-function format_month(dateformat) {
-    var d = new Date(dateformat);
-    return {'v': d, 'f': moment(d).format('MMM YYYY')};
+function check_state_and_draw(filtered_data){
+    var field_name = get_radio_value("radio_button2") + get_radio_value("radio_button1");
+    draw_highcharts_tat_graphs(filtered_data, field_name, get_radio_value("radio_button1") );
 }
 
-// Add a percentage in a list of dict with the provided name
-function add_percentage(dict, numerator, denominator, name){
-    var k;
-    for (k in dict) {
-        dict[k][name] = dict[k][numerator]/dict[k][denominator];
-    }
+function all_tat_charts(data){
+    // Prepare the data for display
+    var finished_data = data.filter(function (d){
```

Comment:
`data` here appears to be a list, and does not have a `filter` method

Reply:
This code has been refactored